src/fork_exec/postfork.rs

```diff
@@ -318,8 +318,8 @@ pub(crate) fn safe_report_exec_error(
                 } else {
                     fstat(fd).map_err(|_| ())
                 };
-                #[allow(clippy::useless_conversion)] // for mode
-                if md.is_err() || unsafe { libc::access(interpreter.as_ptr(), libc::X_OK) } != 0 {
+
+                let err_or_no_exec_handling = || {
```

Comment:
A little worried about this - this is post-fork code, and I can't say with confidence that creating this capturing lambda won't cause a heap allocation. Can we rework this so it's manifestly async-signal safe?